src/compare-tree/key-path.js

```diff
@@ -0,0 +1,40 @@
+const KEY_DELIMETER  = '.'
+const PATH_DELIMETER = ','
+
+import {
+  castPath
+} from '../utils'
+
+function byKeyLength (a, b) {
+  return a.length > b.length ? 1 : -1
+}
+
+export function getKeyString (value) {
+  return value.join(KEY_DELIMETER)
+}
+
+export function getKeyPaths (value) {
+  if (value == null) {
+    return []
+  }
+
+  let paths = value
+
+  if (Array.isArray(value) === false) {
+    paths = `${paths}`.split(PATH_DELIMETER)
+  }
+
+  return paths.map(castPath)
```

Comment:
curious why `castPath` is necessary, seems like you're splitting on the PATH_DELIMETER (`.`), and `castPath` does the same thing. is this covering a case where you pass in an array of items where each item could contain multiple paths (hoping not because that would be a nested array which doesn't sound nice here)